### ISSUE TITLE
Got 500 error on delete method

### ISSUE THREAD

**Issue by DZDeemix** [Sopamo/laravel-filepond#41]
When I try delete file, the $filePath variable equals
`"/var/www/app/storage/app/filepond/rLW7knr1FRSjGkDI/avatar.jpeg"`
But if you use:
`Storage::disk('local')->delete($filePath)` , it should be:
"filepond/rLW7knr1FRSjGkDI/avatar.jpeg"
Is this error?

**Comment by rivalex** [comment]
> When I try delete file, the $filePath variable equals
> `"/var/www/app/storage/app/filepond/rLW7knr1FRSjGkDI/avatar.jpeg"`
> But if you use:
> `Storage::disk('local')->delete($filePath)` , it should be:
> "filepond/rLW7knr1FRSjGkDI/avatar.jpeg"
> Is this error?

Hi @DZDeemix I re-write the controller changing the delete method to make sure that:

- Correctly delete the file
- Remove the empty folder
- Retrive error code on server error

```php
public function delete(Request $request)
{
    try {
        $filePath = $this->filepond->getPathFromServerId($request->getContent());
        $path = pathinfo($filePath, PATHINFO_DIRNAME);
        File::delete($filePath);
        if (count(glob($path . '*')) === 0) {
            rmdir($path);
        }
        return Response::make('', 200, [
            'Content-Type' => 'text/plain',
        ]);
    } catch (Throwable $e) {
        return Response::make($e->getMessage(), 500, [
            'Content-Type' => 'text/plain',
        ]);
    }
}
```

**Comment by DZDeemix** [comment]
`return Response::make('', 200, [
            'Content-Type' => 'text/plain',
        ]);`
I think HTTP code should be 204, when file is deleted. I check this code and it works well. Thanks you.

**Comment by abrardev99** [comment]
Above code through File not found an exception.
Here is the fix

```
public function delete(Request $request)
{
    try {
        $filePath = $this->filepond->getPathFromServerId($request->getContent());
        $path = pathinfo($filePath, PATHINFO_DIRNAME);
        \File::delete($filePath);
        if (count(glob($path . '/*')) === 0) {
            rmdir($path);
        }
        return Response::make('', 200, [
            'Content-Type' => 'text/plain',
        ]);
    } catch (Throwable $e) {
        return Response::make($e->getMessage(), 500, [
            'Content-Type' => 'text/plain',
        ]);
    }
}
```

**Comment by JunichiroNakashima** [comment]
I also encountered this error, and found the reason;

Laravel/Filepond converts the serverId into a full-path, WITH the base.
(like /your-app/storate/app/filepond)

When FilepondController::delete(), 
Laravel tries to add the base folder on top of the full path again,
so files are always not found.


I avoided this error by creating new function

Filepond::getPathFromServerIdWithoutBase

but I hope someone fixes on the source in more elegant way.
